controllers/categoryController.test.js

```diff
@@ -0,0 +1,124 @@
+import slugify from "slugify";
+import categoryModel from "../models/categoryModel";
+import { 
+    createCategoryController, 
+    updateCategoryController, 
+    categoryControlller, 
+    singleCategoryController, 
+    deleteCategoryCOntroller
+} from "./categoryController";
+
+jest.mock("../models/categoryModel");
+jest.mock("slugify");
+
+describe('category controller', () => {
+    let req, res, consoleLogSpy;
+
+    beforeEach(() => {
+        req = {};
+        res = {
+            status: jest.fn().mockReturnThis(),
+            send: jest.fn(),
+        };
+    });
+
+    test('should return 401 if no req body', async () => {
+        req.body = {};
+
+        await createCategoryController(req, res);
+        expect(res.status).toHaveBeenCalledWith(401);
+        expect(res.send).toHaveBeenCalledWith({ message: "Name is required" });
+    })
+
+    test('should return 200 if catergory exists', async () => {
+        req.body = { name: "test" };
+        categoryModel.findOne.mockResolvedValue({ name: "test" });
+
+        await createCategoryController(req, res);
+        expect(res.status).toHaveBeenCalledWith(200);
+        expect(res.send).toHaveBeenCalledWith({ success: true, message: "Category Already Exists" });
+    })
+
+    test('should return 201 and create new category', async () => {
+        req.body = { name: 'test' };
+        categoryModel.findOne.mockResolvedValue(null);
+        slugify.mockReturnValue('test-slug');
+        categoryModel.mockImplementation((data) => ({
+            ...data,
+            save: jest.fn().mockResolvedValue(data)
+        }));
+
+        await createCategoryController(req, res);
+        expect(res.status).toHaveBeenCalledWith(201);
+        expect(res.send).toHaveBeenCalledWith({
+            success: true,
+            message: 'new category created',
+            category: { name: 'test', slug: 'test-slug' }
+        });
+    });
+
+    test('should return 500 and error while creating category', async () => {
+        consoleLogSpy = jest.spyOn(console, 'log').mockImplementation(() => {});
```

Comment:
This `consoleLogSpy` does not seem to be used in any assertions. Did you forget to check it?